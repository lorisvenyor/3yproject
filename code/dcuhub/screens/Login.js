
import React from 'react'
import { StyleSheet, Text, View, Image, Dimensions, TextInput, TouchableOpacity, Alert} from 'react-native'
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { AntDesign } from '@expo/vector-icons'
import { StatusBar } from 'expo-status-bar'
import { SafeAreaView } from 'react-native-safe-area-context'

// import stuff from other files
import SignUpButton from '../shared/button_1'
import LogInButton from '../shared/button_2'
import * as firebase from 'firebase'

const DeviceHeight = Dimensions.get('screen').height

export default class Login extends React.Component{

    state = {
        email: "",
        password: "",
        errorMessage: null
    }

    handleLogin = () => {

        const { email, password } = this.state;
        
        firebase.auth().signInWithEmailAndPassword(email, password ).then(authUser => {
            if(authUser.user.emailVerified){
                console.log('email is verified')
            }else if(!authUser.user.emailVerified){
                console.log('email not verified')
                Alert.alert('email not verified')
                firebase.auth().signOut()
            }
            }).catch(error => this.setState({ errorMessage: error.message}))
    }

    render() {
        return (
            <SafeAreaView style={styles.container}>
            <StatusBar style="auto"/>
            <KeyboardAwareScrollView>

                <View style={styles.image_view}>  
                    <Image 
                    source={require("../assets/images/dcuhublogo_higherres.png")}
                    resizeMode="contain"
                    resizeMode="center"
                    style={styles.image_style}>
                    </Image>
                </View> 

                <Text style={styles.welcome_txt}> Welcome back, Student!</Text> 
                
                {/* firebase errors should pop up here */}
                <View style={styles.errorMessage}>
                {this.state.errorMessage && <Text style={styles.error_txt}>{this.state.errorMessage}</Text>}
                </View>


                {/* this is for the email address part */}
                <View style={styles.email_add}>
                    <AntDesign name="mail" color="#25afcd" size={25}/>
                    <TextInput 
                        style={styles.placeholder_text}
                        placeholder="enter your dcu email address"
                        onChangeText={email => this.setState({email})}
                        value={this.state.email}>
                    </TextInput>
                </View>
            
                {/* this is for the password part */}
                <View style={styles.pass_word}>
                    <AntDesign name="lock1" color="#25afcd" size={25}/>
                    <TextInput 
                        style={styles.placeholder_text}
                        secureTextEntry={true}  // SECURE PASSWORD ie hides the texts
                        placeholder="enter your password"
                        autoCapitalize="none"
                        onChangeText={password => this.setState({password})}
                        value={this.state.password}
                        ></TextInput>
                </View>
            
                {/* THIS IS A BUTTON THAT NAVIGATES TO THE NEWSFEED SECTION - LINKS TO FIREBASE AUTHENTICATION */}
                <View style={{marginTop: 20}}>
                    <LogInButton text='LOG IN' onPress={this.handleLogin}/>
                </View>

                {/* THIS IS A BUTTON THAT NAVIGATES TO THE REGISTER SECTION/SCREEN  */}
                <View>
                    <SignUpButton text='SIGN UP' onPress={() => this.props.navigation.navigate('Register')}/>
                </View>
                
                <TouchableOpacity style={styles.forgot_pass} onPress={() => this.props.navigation.navigate('Reset')}>
                    <Text style={{color: "#d40059", fontSize: 18}}>Forgot Password?</Text>
                </TouchableOpacity>
        
            
            </KeyboardAwareScrollView>
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container : {
        flex: 1,
        height: DeviceHeight,
        backgroundColor: "white"
    },
    email_add: {
        flexDirection:"row",
        alignItems:"center",
        marginHorizontal:40,
        marginTop: 20,
        paddingHorizontal:10,
        paddingVertical: 7,
        borderWidth: 1.5,
        borderColor: "#25afcd",
        borderRadius: 10,
    },
    pass_word: {
        flexDirection:"row",
        alignItems:"center",
        marginHorizontal:40,
        borderWidth: 1.5,
        marginTop:15,
        paddingHorizontal:10,
        borderColor:"#25afcd",
        borderRadius:10,
        paddingVertical: 7,
    },
    errorMessage: {
        alignItems: "center",
        justifyContent: "center",
        marginHorizontal: 30,
        marginTop: 30
    },
    error_txt:{
        color: "#d40059",
        fontSize: 15,
        fontWeight: "600",
        textAlign: "center"
    },

    image_style: {
        width: Dimensions.get('screen').width, 
        height: Dimensions.get('screen').height / 2
    },

    image_view: {
        alignSelf: 'center',
        justifyContent: 'center',
        marginTop: -40

    },
    welcome_txt: {
        fontSize: 25, 
        alignSelf: "center",
        marginTop: -50,
        fontWeight: "300"
    },
    placeholder_text: {
        paddingHorizontal: 10
    }, 

    forgot_pass: {
        marginTop: 10,
        alignSelf: 'center',
        marginBottom: 30
    }
})

// colours 
// blue #25afcd
// orange #f4a446
// pink #d40059