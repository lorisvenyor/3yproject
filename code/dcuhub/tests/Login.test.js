import React from 'react'
import {create} from 'react-test-renderer'
import Login from '../screens/Login'

jest.useFakeTimers()

const screen = create(<Login/>)

test('Login Screen Snapshot', () => {
    expect(screen).toMatchSnapshot()
})

