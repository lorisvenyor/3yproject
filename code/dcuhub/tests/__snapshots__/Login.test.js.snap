// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen Snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "marginTop": -40,
          }
        }
      >
        <Image
          resizeMode="center"
          source={1}
          style={
            Object {
              "height": 667,
              "width": 750,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 25,
            "fontWeight": "300",
            "marginTop": -50,
          }
        }
      >
         Welcome back, Student!
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 30,
            "marginTop": 30,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "marginHorizontal": 40,
            "marginTop": 20,
            "paddingHorizontal": 10,
            "paddingVertical": 7,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="enter your dcu email address"
          rejectResponderTermination={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#25afcd",
            "borderRadius": 10,
            "borderWidth": 1.5,
            "flexDirection": "row",
            "marginHorizontal": 40,
            "marginTop": 15,
            "paddingHorizontal": 10,
            "paddingVertical": 7,
          }
        }
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="enter your password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "orange",
                "borderRadius": 20,
                "justifyContent": "center",
                "marginHorizontal": 100,
                "marginTop": 30,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              LOG IN
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#f4a446",
                "borderRadius": 20,
                "borderWidth": 2,
                "justifyContent": "center",
                "marginHorizontal": 100,
                "marginTop": 10,
                "paddingVertical": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#f4a446",
                  "fontSize": 20,
                }
              }
            >
              SIGN UP
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 30,
            "marginTop": 10,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#d40059",
              "fontSize": 18,
            }
          }
        >
          Forgot Password?
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
